c10/cuda/CMakeLists.txt

```diff
@@ -70,7 +70,7 @@ if(NOT BUILD_LIBTORCHLESS)
 
   if(NOT WIN32)
   target_link_libraries(c10_cuda PRIVATE dl)
-  target_compile_options(c10_cuda PRIVATE "-DPYTORCH_C10_DRIVER_API_SUPPORTED")
```

Comment:
This is needed in order for `c10d`, which depends on `c10_cuda`, to be able to check the value of the `PYTORCH_C10_DRIVER_API_SUPPORTED` macro. However, I suspect there was a reason for this to be set to private, hence if it's a constraint to keep it so we'll have to find another way.

Reply:
What diff does PRIVATE make compared to PUBLIC?

Reply:
PUBLIC makes it so that every target that depends on `c10_cuda` has this macro defined. It may thus "pollute" a lot of downstream targets, perhaps even user code?

I don't know who's the expert these days on the PyTorch build system, let me tag @malfet to see if he has any opinions/tips?

Reply:
I found a way to avoid changing it to PUBLIC: there's a list of files for the distributed package on which we set that preprocessor macro, but the cuda/utils.cpp one was missing.